apps/server/src/controllers/v1/events.ts

```diff
@@ -44,6 +44,86 @@ router.post('/', verifyToken, async (req: Request, res: Response) => {
     res.sendStatus(204)
 })
 
+router.get('/search', async (req: Request, res: Response) => {
+    const { q, category } = req.query
+    const target = (req.query.target as string) || 'default'
+    const sort = req.query.sort || 'default'
+    const startDate = req.query.startDate ? new Date(req.query.startDate as string) : new Date()
+    const endDate = req.query.endDate ? new Date(req.query.endDate as string) : undefined
+    const isAllDay = req.query.isAllDay === 'true'
+    const page = Number(req.query.page) || 1
+    const limit = Number(req.query.limit) || 10
+    const skip = (page - 1) * limit
+
+    const pathMap: Record<string, string[]> = {
+        default: ['name', 'description'],
+        name: ['name'],
+        address: ['address'],
+    }
+
+    const path = pathMap[target]
+
+    if (!q) {
+        throw new KeywordIsEmptyToSearch()
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const query: any[] = [
+        {
+            $search: {
+                index: 'korean_events_search',
+                text: {
+                    query: q,
+                    path: path,
+                },
+            },
+        },
+        { $skip: skip },
+        { $limit: limit },
+    ]
+
+    if (category) {
+        query.push({
+            $match: { category },
+        })
+    }
+
+    if (startDate || endDate) {
+        if (startDate > endDate) {
```

Comment:
```suggestion
    if (endDate && startDate > endDate) {
```

이렇게 변경해서 error를 던지게 하는건 어떨까요? `||`로 존재 여부를 검증하는데, 아래 내부 요소는 `&&`로 검증하고 있어서 두번째 nested-if가 없어도 될 것 같아서요